template/Vagrantfile.erb

```diff
@@ -15,20 +15,22 @@ Vagrant.configure('2') do |config|
 
   ## EC2 Provider
   config.vm.provider '<%= ec2.provider %>' do |ec2, override|
+    override.vm.box = '<%= ec2.box %>'
+    override.vm.box_url = '<%= ec2.box_url %>'
+    override.ssh.username = '<%= ec2.ssh_username %>'
+
+    ec2.region = '<%= ec2.region %>'
     ec2.subnet_id = '<%= ec2.subnet_id %>'
-    ec2.security_groups = <%= ec2.security_groups %>
-    ec2.iam_instance_profile_arn = '<%= ec2.instance_profile %>'
 
-<% unless ec2.public_ip -%>
     ## VPN Connected VPC
-    ec2.associate_public_ip = false
-    ec2.ssh_host_attribute = :private_ip_address
-<% end -%>
-    ec2.region = '<%= ec2.region %>'
+    ec2.associate_public_ip = <%= ec2.associate_public_ip %>
+    ec2.ssh_host_attribute = :<%= ec2.ssh_host_attribute %>
+
     ec2.instance_type = '<%= ec2.instance_type %>'
+    ec2.security_groups = <%= ec2.security_groups %>
+    ec2.iam_instance_profile_arn = '<%= ec2.iam_instance_profile_arn %>'
```

Comment:
From what I understand these settings take precedence over the settings in `~/.vagrant.d/Vagrantfile`[1].  Do you expect the `~/.vagrant.d/Vagrantfile` and the `~/.builderator/Buildfile` to be mutually exclusive?

[1] https://docs.vagrantup.com/v2/vagrantfile/